htdocs/core/modules/dons/html_cerfafr.modules.php

```diff
@@ -297,6 +297,10 @@ private function amountToLetters($montant, $devise1 = '', $devise2 = '')
 		$prim = array();
 		$secon = array();
 		$trio = array();
+		'@phan-var string[] $prim
```

Comment:
Did you try to use
`/** var string[] $prim */`
before the $prim = array(); 
So this avoid to introduce phan specific code. Dos this trick works ?

Reply:
I use phan in other projects - as far as I am used to using PHPDoc does not work here for phan, so I do not try '/** @var ... */ anymore when the case corresponds to something where that would not work.

The documentation explains that this is a limition of AST:
https://github.com/phan/phan/wiki/Annotating-Your-Source-Code#inline-type-checks-via-string-literals

I try to avoid specific code as much as possible myself too.